packages/app-store/zohocalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,412 @@
+
+    return fetch(`https://calendar.zoho.com/api/v1${endpoint}`, {
+      method: "GET",
+      ...init,
+      headers: {
+        Authorization: "Bearer " + credentials.access_token,
+        "Content-Type": "application/json",
+        ...init?.headers,
+      },
+    });
+  };
+
+  private getUserInfo = async () => {
+    const credentials = await this.auth.getToken();
+
+    const response = await fetch(`https://accounts.zoho.com/oauth/user/info`, {
+      method: "GET",
+      headers: {
+        Authorization: "Bearer " + credentials.access_token,
+        "Content-Type": "application/json",
+      },
+    });
+
+    return this.handleData(response, this.log);
+  };
+
+  async createEvent(event: CalendarEvent): Promise<NewCalendarEventType> {
+    let eventId = "";
+    let eventRespData;
+    const [mainHostDestinationCalendar] = event.destinationCalendar ?? [];
+    const calendarId = mainHostDestinationCalendar?.externalId;
+    if (!calendarId) {
+      throw new Error("no calendar id");
+    }
+
+    try {
+      const query = stringify({
+        eventdata: JSON.stringify(this.translateEvent(event)),
+      });
+
+      const eventResponse = await this.fetcher(`/calendars/${calendarId}/events?${query}`, {
+        method: "POST",
+      });
+      eventRespData = await this.handleData(eventResponse, this.log);
+      eventId = eventRespData.events[0].uid as string;
+    } catch (error) {
+      this.log.error(error);
+      throw error;
+    }
+
+    try {
+      return {
+        ...eventRespData.events[0],
+        uid: eventRespData.events[0].uid as string,
+        id: eventRespData.events[0].uid as string,
+        type: "zoho_calendar",
+        password: "",
+        url: "",
+        additionalInfo: {},
+      };
+    } catch (error) {
+      this.log.error(error);
+      await this.deleteEvent(eventId, event, calendarId);
+      throw error;
+    }
+  }
+
+  /**
+   * @param uid
+   * @param event
+   * @returns
+   */
+  async updateEvent(uid: string, event: CalendarEvent, externalCalendarId?: string) {
+    const eventId = uid;
+    let eventRespData;
+    const [mainHostDestinationCalendar] = event.destinationCalendar ?? [];
+    const calendarId = externalCalendarId || mainHostDestinationCalendar?.externalId;
+    if (!calendarId) {
+      this.log.error("no calendar id provided in updateEvent");
+      throw new Error("no calendar id provided in updateEvent");
+    }
+    try {
+      // needed to fetch etag
+      const existingEventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}`);
+      const existingEventData = await this.handleData(existingEventResponse, this.log);
+
+      const query = stringify({
+        eventdata: JSON.stringify({
+          ...this.translateEvent(event),
+          etag: existingEventData.events[0].etag,
+        }),
+      });
+
+      const eventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}?${query}`, {
```

Comment:
<img width="757" alt="Screenshot 2023-09-25 at 1 10 14 PM" src="https://github.com/calcom/cal.com/assets/1780212/7765925c-2674-4087-a3b2-6d24f139e644">
If for the booking being rescheduled in Cal.com, Zoho calendar event doesn't exist due to some reason(I deleted that in my case), there comes an error in rescheduling the event on Zoho.

I think we should handle it gracefully, by creating the event on rescheduled time.
Though not a blocker, we can create a ticket for it and work on it later.